artichoke-backend/src/extn/core/symbol/mod.rs

```diff
@@ -1,115 +1,14 @@
 use std::ffi::c_void;
 
 use crate::convert::{Immediate, UnboxedValueGuard};
-use crate::extn::core::array::Array;
 use crate::extn::prelude::*;
-use crate::intern::Symbol as SymbolId;
 
 pub mod ffi;
 pub mod mruby;
 pub mod trampoline;
 
-#[derive(Debug, Clone, Copy, Hash, PartialEq, Eq, PartialOrd, Ord)]
-pub struct Symbol(SymbolId);
-
-impl Symbol {
-    pub fn all_symbols(interp: &mut Artichoke) -> Result<Array, Exception> {
-        let all_symbols = interp.all_symbols()?.map(Symbol::from).collect::<Vec<_>>();
-        let array = all_symbols
-            .into_iter()
-            .map(|symbol| Symbol::alloc_value(symbol, interp))
-            .collect::<Result<Vec<Value>, _>>()?;
-        Ok(Array::from(array))
-    }
-
-    #[inline]
-    #[must_use]
-    pub fn id(self) -> SymbolId {
-        self.0
-    }
-
-    #[must_use]
-    pub fn is_empty(self, interp: &mut Artichoke) -> bool {
```

Comment:
these methods for manipulating byte content were mostly lifted directly into `spinoso_symbol::Symbol` and hidden behind the `artichoke` Cargo feature.

One nice change resulting from that is most APIs only take `&T where T: Intern`. The `&mut` requirement here is likely a copy-paste oversight.